chainer/functions/loss/mean_absolute_error.py

```diff
@@ -19,22 +19,24 @@ def check_type_forward(self, in_types):
         )
 
     def forward_cpu(self, inputs):
+        self.retain_inputs((0, 1))
         x0, x1 = inputs
-        self.diff = x0 - x1
-        diff = self.diff.ravel()
+        diff = (x0 - x1).ravel()
         return numpy.array(abs(diff).sum() / diff.size, dtype=diff.dtype),
 
     def forward_gpu(self, inputs):
+        self.retain_inputs((0, 1))
         x0, x1 = inputs
-        self.diff = x0 - x1
-        diff = self.diff.ravel()
+        diff = (x0 - x1).ravel()
         return abs(diff).sum() / diff.dtype.type(diff.size),
 
     def backward(self, indexes, grad_outputs):
+        x0, x1 = self.get_retained_inputs()
+        diff = x0 - x1
         gy, = grad_outputs
-        coeff = gy * gy.data.dtype.type(1. / self.diff.size)
-        coeff = chainer.functions.broadcast_to(coeff, self.diff.shape)
-        gx0 = coeff * cuda.get_array_module(gy.data).sign(self.diff)
+        coeff = gy * gy.data.dtype.type(1. / diff.size)
```

Comment:
gy.data -> gy.array?

Reply:
According to [this doc](https://github.com/chainer/chainer/blob/master/chainer/variable.py#L655-L656), `.array` is better. However, it should be done in another PR because it's not related to this bug (I know I changed one `data` to `array`).

@kmaehashi Do I need to replace all `data` with `array` in this PR?

Reply:
> Do I need to replace all data with array in this PR?

You don't need to do that in this PR. Let's focus on fixing bugs.